bodhi/tests/server/consumers/test_signed.py

```diff
@@ -43,3 +44,69 @@ def test_calls_super(self, __init__):
 
         self.assertEqual(handler.topic, ['topic_prefix.environment.buildsys.tag'])
         __init__.assert_called_once_with(hub)
+
+
+class TestSignedHandlerConsume(unittest.TestCase):
+    """Test class for the :func:`SignedHandler.consume` method."""
+
+    def setUp(self):
+        self.sample_message = {
+            'body': {
+                'i': 628,
+                'timestamp': 1484692585,
+                'msg_id': '2017-821031da-be3a-4f4b-91df-0baa834ca8a4',
+                'crypto': 'x509',
+                'topic': 'org.fedoraproject.prod.buildsys.tag',
+                'signature': '100% real please trust me',
+                'msg': {
+                    'build_id': 442562,
+                    'name': 'colord',
+                    'tag_id': 214,
+                    'instance': 's390',
+                    'tag': 'f26-updates-testing-pending',
+                    'user': 'sharkcz',
+                    'version': '1.3.4',
+                    'owner': 'sharkcz',
+                    'release': '1.fc26'
+                },
+            },
+        }
+        hub = mock.MagicMock()
+        hub.config = {'environment': 'environment', 'topic_prefix': 'topic_prefix'}
+        self.handler = signed.SignedHandler(hub)
+
+    @mock.patch('bodhi.server.consumers.signed.Build')
+    def test_consume(self, mock_build_model):
+        """Assert that messages marking the build as signed updates the database"""
+        build = mock_build_model.get.return_value
+        build.release.pending_testing_tag = self.sample_message['body']['msg']['tag']
+
+        self.handler.consume(self.sample_message)
+        self.assertTrue(build.signed is True)
+
+    @mock.patch('bodhi.server.consumers.signed.Build')
+    def test_consume_not_pending_testing_tag(self, mock_build_model):
+        """Assert that messages marking the build as signed updates the database"""
+        build = mock_build_model.get.return_value
+        build.release.pending_testing_tag = "some tag that isn't pending testing"
+
+        self.handler.consume(self.sample_message)
+        self.assertFalse(build.signed is True)
+
+    @mock.patch('bodhi.server.consumers.signed.Build')
+    def test_consume_no_release(self, mock_build_model):
+        """Assert that messages marking the build as signed updates the database"""
```

Comment:
This docblock should probably say that it's checking that builds on releases that don't specify the tag get marked as signed.